internal/server/v1beta1/server.go

```diff
@@ -28,23 +28,27 @@ type APIServer struct {
 
 var errMissingUserInfo = errors.New("missing user information")
 
-func NewAPIServer(
-	logger log.Logger,
-	assetService AssetService,
-	starService StarService,
-	discussionService DiscussionService,
-	tagService TagService,
-	tagTemplateService TagTemplateService,
-	userService UserService,
-) *APIServer {
+type APIServerDeps struct {
+	Asset       AssetService
+	Star        StarService
+	Discussion  DiscussionService
+	Tag         TagService
+	TagTemplate TagTemplateService
+	User        UserService
+	Logger      log.Logger
+	StatsD      StatsDClient
+}
+
+func (d APIServerDeps) Build() *APIServer {
```

Comment:
[just nit] the commonly used convention in Go is to use `New` as the prefix for constructor functions, unless we have strong needs of using factory pattern,

let's keep the old style with replacing it with deps struct
```go
func NewAPIServer(deps APIServerDeps) *APIServer {}